docs/Threading.md

```diff
@@ -0,0 +1,167 @@
+The SGX-LKL threading implementation
+====================================
+
+SGX-LKL has four kinds of threads: ethreads, lthreads, Linux tasks, and pthreads.
+This document explains how they are implemented and how they relate.
+
+Ethreads are the lowest-level threads are ethreads.
+These are roughly analogous to VCPUs in conventional VMs.
+In SGX, each ethread corresponds to a thread-control structure (TCS), allocated by OpenEnclave.
+
+The lthreads layer provides a cooperative threading library on top of ethreads.
```

Comment:
lthreads are some library https://lthread.readthedocs.io/en/latest/intro.html ? The name confuses me, l is not for linux.

Reply:
It's based on that library, but heavily modified.

Reply:
Mentioning that will help people understand that l is not for linux and that they are on their own anyway. I guess ethreads are enclave threads too, implied but not stated. It may be me being thick or that it is such a long time since I did guts level linux stuff.

Reply:
```suggestion
The lthreads (lightweight threads) layer provides a cooperative threading library on top of ethreads.
```